runtime/expr/filter.go

```diff
@@ -113,6 +121,31 @@ func (s *search) Eval(_ Context, this *zed.Value) *zed.Value {
 	return zed.False
 }
 
+type searchCIDR struct {
+	net   *net.IPNet
+	bytes zcode.Bytes
+}
+
+func (s *searchCIDR) Eval(_ Context, val *zed.Value) *zed.Value {
+	if errMatch == val.Walk(func(typ zed.Type, body zcode.Bytes) error {
+		switch typ.ID() {
+		case zed.IDNet:
+			if bytes.Compare(body, s.bytes) == 0 {
+				return errMatch
+			}
+		case zed.IDIP:
+			addr := zed.DecodeIP(body).IPAddr().IP
```

Comment:
`netaddr.IP.IPAddr` allocates memory. Replacing the `*net.IPNet` with a `netaddr.IPPrefix` would avoid that.